menuconfig.py

```diff
@@ -165,100 +165,266 @@
+        "separator":        "fg:black,bg:yellow,bold",
+        # List of items, e.g. the main display
+        "list":             "fg:black,bg:white",
+        # Style for the selected item
+        "selection":        "fg:white,bg:blue,bold",
+        # Like _LIST_(SEL_)STYLE, for invisible items. Used in show-all mode.
+        "inv_list":         "fg:red,bg:white",
+        "inv_selection":    "fg:red,bg:white",
+        # Help text windows at the bottom of various fullscreen dialogs
+        "help":             "fg:black,bg:white,bold",
+        # Frame around dialog boxes
+        "frame":            "fg:black,bg:yellow,bold",
+        # Body of dialog boxes
+        "body":             "fg:white,bg:black",
+        # Edit box in pop-up dialogs
+        "edit":             "fg:white,bg:blue",
+        # Edit box in jump-to dialog
+        "jump_to_edit":     "fg:white,bg:blue",
+        # Symbol information text
+        "text":             "fg:black,bg:white"
+}
+
+# This style is forced on terminals that do no support colors
+_MONOCHROME_STYLE = \
+{
+        "path":             "bold",
+        "separator":        "bold,standout",
+        "list":             "",
+        "selection":        "bold,standout",
+        "inv_list":         "bold",
+        "inv_selection":    "bold,standout",
+        "help":             "bold",
+        "frame":            "bold,standout",
+        "body":             "",
+        "edit":             "standout",
+        "jump_to_edit":     "",
+        "text":             ""
+}
+
+_AQUATIC_STYLE = \
+{
+        "style":            "default",
+        "path":             "fg:cyan,bg:blue,bold",
+        "separator":        "fg:white,bg:cyan,bold",
+        "help":             "fg:cyan,bg:blue,bold",
+        "frame":            "fg:white,bg:cyan,bold",
+        "body":             "fg:white,bg:blue",
+        "edit":             "fg:black,bg:white",
+}
+
+_STYLES = {}
+_CUSTOM_STYLE = None
+
+class _StyleError(Exception):
+    pass
+
+def _style_expand(sdef, base = None):
+    global _STYLES
+    global _DEFAULT_STYLE
+
+    # base must be an already expanded style
+    if "style" in sdef:
+        if not sdef["style"] in _STYLES:
+            raise _StyleError("Unkown style: {}".format(sdef["style"]))
+
+        style = dict(_STYLES[sdef["style"]])
+    elif base:
+        style = dict(base)
+    else:
+        style = dict(_DEFAULT_STYLE)
 
-    # Initialize styles for different parts of the application. The arguments
-    # are ordered as follows:
-    #
-    #   1. Text color
-    #   2. Background color
-    #   3. Attributes
-    #   4. Extra attributes if colors aren't available. The colors will be
-    #      ignored in this case, and the attributes from (3.) and (4.) will be
-    #      ORed together.
+    # Merge the two dictionaries
+    for x in sdef:
+        # Skip the "style" special keyword
+        if x == "style": continue
```

Comment:
Wouldn't be any need for the `style` keyword with the string-only design I think.

Reply:
The string representation of a style (as defined, for example, in the environment) is converted to a dictionary type. The "style" keyword is a dictionary keyword -- it is used when there's a word without '=' in it in the string.

So for example:

MENUCONFIG_STYLE="aquatic"

is converted to:

{
      "style": "aquatic"
}

Probably this could be renamed to "template" to make it clearer what the intention is.